serverless/src/tracing.ts

```diff
@@ -48,7 +48,10 @@ export enum TracingMode {
   NONE,
 }
 
-function findIamRole(resources: Resources, lambda: LambdaFunction): IamRoleProperties | undefined {
+function findIamRoleFromCloudFormationTemplate(
```

Comment:
Renamed `findIamRole` to `findIamRoleFromCloudFormationTemplate`, and extracted two functions:
1. `enableXrayTracing`
2. `enableDatadogTracing`